numerics/fast_fourier_transform.hpp

```diff
@@ -0,0 +1,47 @@
+
+#pragma once
+
+#include <array>
+#include <complex>
+#include <vector>
+
+#include "base/bits.hpp"
+#include "quantities/named_quantities.hpp"
+
+namespace principia {
+namespace numerics {
+namespace internal_fast_fourier_transform {
+
+using base::FloorLog2;
+
+// This class computes Fourier[{...}, FourierParameters -> {1, -1}] in
+// Mathematica notation.  (The "signal processing" Fourier transform.)
+template<typename Container, int size_>
```

Comment:
We do not use Container outside the constructors: templatize on `Scalar` instead, templatize the constructors.

Reply:
Template parameter sizes that interact with containers have to be `std::size_t`, otherwise deductions break down.